### PR TITLE
169/proposal for new crawling website int

### PR DESCRIPTION
**Pull request by DanielKraft**
You can test it with the following commands:

```shell
$ cd etl
$ python (or python3)
>>> from data_extraction.scraper.weltwaerts import WeltwaertsScraper as ww
>>> i = ww('test')
>>> i.add_urls()
(approx. 2 min)

>>> i.crawl(i.urls[0], 1)
>>> i.data
```

### CODE REVIEW

**Review by MaiThiD — COMMENTED** [review]
"We have decided to leave out logos. So:

1.	Remove from data model
2.	Do not set in backend"(https://github.com/hochschule-darmstadt/einander-helfen/issues/189#issuecomment-735798346)

**Comment by DanielKraft** [comment]
> "We have decided to leave out logos. So:
> 
> 1.	Remove from data model
> 2.	Do not set in backend"(https://github.com/hochschule-darmstadt/einander-helfen/issues/189#issuecomment-735798346)
> 

Done.

File: etl/data_extraction/scraper/weltwaerts.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    def add_urls(self):
+        import time
+
+        search_page_url = f'{self.base_url}/de/ep-ergebnis.html'
+        next_page_url = search_page_url
+
+        index = 1
+        while next_page_url:
+
+            response = self.soupify(next_page_url)
```
Comment (FiedlerJan) [inline etl/data_extraction/scraper/weltwaerts.py:101]: Currently you have only 10 results per page. The page gives you an option to scrape up to 50 results per page. This is possible by sending this arument with the request. Is there a possible implementation? (Maybe by overloading the soupify method which takes and request argument)
grafik

**Review by FiedlerJan — CHANGES_REQUESTED** [review]
I got same more things you should take a look on:

- Please write the class and function description in a docstring format ("""description""")
- Why are there so many errors due to duplicated links?

File: etl/data_extraction/scraper/weltwaerts.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    #   Adds all URLs of detail pages, found on the search pages, for the crawl function to scrape
+    def add_urls(self):
+        import time
+
+        search_page_url = f'{self.base_url}/de/ep-ergebnis.html'
+        next_page_url = search_page_url
+
+        index = 1
+        while next_page_url:
```
[the remark this answers is not in the document]
Comment (DanielKraft) [reply etl/data_extraction/scraper/weltwaerts.py:99]: The maximum number of pages is not necessary for our implementation.
What is the advantage of using the maximum number of pages?

**Comment by DanielKraft** [comment]
> * Why are there so many errors due to duplicated links?

For some reason sometimes entries are listed twice. The error is not due to the crawler, but to weltwärts.de

File: etl/data_extraction/scraper/weltwaerts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FiedlerJan) [reply etl/data_extraction/scraper/weltwaerts.py:103]: I think there is no direct advantage. Only thing is that the debugging is more obvious (f.e. 4/567).

[the remark this answers is not in the document]
Comment (DanielKraft) [reply etl/data_extraction/scraper/weltwaerts.py:103]: I have added the maximum number of pages to the debug message.

**Review by FiedlerJan — CHANGES_REQUESTED** [review]
Sorry for the late review. I crawled the whole page. During this I found one reoccurring mistake. Somewhere during the crawling process an index is out of range.

```
{
        "fn": "parse",
        "body": "list index out of range",
        "index": 3361,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201020"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3362,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201021"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3363,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201024"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3378,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201023"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3379,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=209700"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3423,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201025"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3435,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201026"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3485,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=209699"
    },
    {
        "fn": "parse",
        "body": "list index out of range",
        "index": 3512,
        "url": "https://www.weltwaerts.de/de/ep-detail.html?id=201019"
    }
```

Please take a look on this.

**Review by FiedlerJan — APPROVED** [review]

**Review by MaiThiD — APPROVED** [review]

**Review by jonas-hda — APPROVED** [review]

**Pull request merged**